### PR TITLE
Better PNG encoding

### PR DESCRIPTION
**Pull request by directionless**

### CODE REVIEW

File: cross_language_tests/png.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+args = ARGV
+
+cmd = args.shift
+pngfile = args.shift # File.read(args.shift)
+outfile = args.shift
+
+case cmd
+when "decode"
+  data = Krypto::Png.decode(pngfile)
```
Comment (directionless) [inline cross_language_tests/png.rb:15]: need to handle streams here

File: krypto.gemspec [hunk 2]
```diff
@@ -19,5 +19,6 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency "openssl"
   s.add_runtime_dependency "msgpack", "~> 1.5"
+  s.add_runtime_dependency "chunky_png"
```
Comment (directionless) [inline krypto.gemspec:22]: Probably pin this?

File: lib/krypto/png.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        ChunkyPNG::Color.r(value),
+        ChunkyPNG::Color.g(value),
+        ChunkyPNG::Color.b(value)
+      ]
+    end
+
+    def self.decode0(pixels)
+      # This uses the chunky_png integer weirdness and bitshifts out our 24bit number.
+      data_len = pixels[1] >> 8
+      pixels_needed = divCeil(data_len, USABLE_BYTES_PER_PIXEL)
```
Comment (levicole) [inline lib/krypto/png.rb:58]: If you want to do this without the helper function:

```
data_len.fdiv(USABLE_BYTES_PER_PIXEL).ceil
```

Will do the same thing!

**Review by levicole — COMMENTED** [review]
I'm good with this. Ruby code looks good. I'm fine with the Go code, but wasn't sure if you wanted @James-Pickett to take a look at it too before final approval.

**Review by James-Pickett — APPROVED** [review]

File: lib/krypto/png.rb (hunk 3 above)
Comment (directionless) [reply lib/krypto/png.rb:58]: I'd opted to avoid floating point. Maybe needless.

File: lib/krypto/version.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module Krypto
-  VERSION = "0.2.1"
+  VERSION = "0.3.1"
```
Comment (directionless) [inline lib/krypto/version.rb:4]:
```suggestion
  VERSION = "0.3.0"
```

**Review by levicole — APPROVED** [review]

**Pull request merged**